django/core/servers/basehttp.py

```diff
@@ -180,6 +182,24 @@ def address_string(self):
         # Short-circuit parent method to not call socket.getfqdn
         return self.client_address[0]
 
+    def log_request(self, code="-", size="-"):
```

Comment:
Should we implement this via `log_request()`? :thinking: I was thinking that changes to `log_messages()` and `ServerFormatter` are needed instead. I'm concerned that overriding `log_request()` may completely ignore custom formatters for `django.server` logger.

Reply:
https://github.com/python/cpython/blob/main/Lib/http/server.py#L545 - `log_request` seemed to be the only place which was creating a `format string` for the logger, so I went ahead with overriding it and appending the `format_string` for a placeholder for response time. 

Another way was to append the `format string` passed in `log_message` with response time, but the former approach seemed better. I can edit the PR to use `log_message` as well.

PS: I tried setting up a simple custom formatter (I have not played with them before) and it seemed to be working fine (Custom formatted message followed by status_code, bytes sent and response time). How should I test this completely to make sure custom formatters are not broken 🤔? ( I will try finding some tests for them)

Reply:
Sorry for the late reply. I'm not sure what's best here - I'm not especially familiar with all the relevant logging features here.

Perhaps the way forward would be to show the code changes that would be required for the different options to see what would be involved?